org.civicrm.reqangular/test/services/check-permission_test.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-env amd, jasmine */
+
+(function (CRM) {
+  define([
+    'common/angular',
+    'common/angularMocks',
+    'common/services/check-permission'
+  ], function () {
+    'use strict';
+
+    describe('checkPermissions', function () {
+      var $rootScope, checkPermissions, promise;
+      var ownedPermissions = ['permission #1', 'permission #2'];
+
+      beforeEach(module('common.services'));
+      beforeEach(inject(function (_$rootScope_, _checkPermissions_) {
+        $rootScope = _$rootScope_;
+        checkPermissions = _checkPermissions_;
+
+        spyOn(CRM, 'checkPerm').and.callFake(function (permission) {
+          return ownedPermissions.indexOf(permission) !== -1;
+        });
+      }));
+
+      afterEach(function () {
+        $rootScope.$digest();
+      });
+
+      describe('basic tests', function () {
+        var permissionToCheck = 'permission #1';
+
+        beforeEach(function () {
+          promise = checkPermissions(permissionToCheck);
+        });
+
+        it('returns a promise', function () {
+          expect(promise).toEqual(jasmine.objectContaining({
+            then: jasmine.any(Function),
+            catch: jasmine.any(Function),
+            finally: jasmine.any(Function)
+          }));
+        });
+
+        it('utilizes the the `CRM.checkPerm` method', function () {
+          expect(CRM.checkPerm).toHaveBeenCalledWith(permissionToCheck);
```

Comment:
@bmnepali @igorpavlov @deb1990 an example of a unit test testing the actual implementation of a method. Usually done to put the emphasis not only on the *what*, but also on an important part of the *how* (which is considered so crucial to deserve to be explicitly tested. For example making sure that a Model service calls the expected API service)